### PR TITLE
Coverity Scan

### PR DESCRIPTION
**Pull request by spurohitsynopsys**

### CODE REVIEW

File: synopsys-task/src/synopsys-task/input.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import * as constants from "./application-constant";
 
 //Bridge download url
 export const BRIDGE_DOWNLOAD_URL =
-  taskLib.getInput("bridge_download_url") || "";
+  "https://sig-repo.synopsys.com/artifactory/bds-integrations-release/com/synopsys/integration/synopsys-bridge/0.1.244/synopsys-bridge-0.1.244-macosx.zip";
```
Comment (kishorikumar) [inline synopsys-task/src/synopsys-task/input.ts:6]: Remove this hardcoded url

File: synopsys-task/src/synopsys-task/input.ts [hunk 2]
```diff
@@ -22,3 +22,26 @@ export const POLARIS_ASSESSMENT_TYPES = taskLib.getDelimitedInput(
 );
 export const POLARIS_SERVER_URL =
   taskLib.getInput(constants.POLARIS_SERVER_URL_KEY) || "";
+
+// Coverity related inputs
```
Comment (kishorikumar) [inline synopsys-task/src/synopsys-task/input.ts:26]: Is this comments required?

File: synopsys-task/src/synopsys-task/validator.ts [hunk 3]
```diff
@@ -79,3 +81,42 @@ export function validateBridgeUrl(url: string): boolean {
...
+  return errors;
+}
+
+export function validateCoverityInstallDirectoryParam(
+  installDir: string
+): boolean {
+  if (
+    installDir != null &&
+    installDir.length > 0 &&
+    !fs.existsSync(installDir)
```
Comment (kishorikumar) [inline synopsys-task/src/synopsys-task/validator.ts:114]: Can we try using tasklib's exist lib?

**Review by kishorikumar — CHANGES_REQUESTED** [review]

File: synopsys-task/src/synopsys-task/input.ts (hunk 2 above)
Comment (spurohitsynopsys) [reply synopsys-task/src/synopsys-task/input.ts:26]: following same strategy as polaris, we have comments above

File: synopsys-task/src/synopsys-task/input.ts (hunk 1 above)
Comment (spurohitsynopsys) [reply synopsys-task/src/synopsys-task/input.ts]: removed

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 4]
```diff
@@ -123,8 +126,18 @@ export class SynopsysBridge {
...
+      const coverityErrors: string[] = validateCoverityInputs();
+      if (coverityErrors.length === 0 && inputs.COVERITY_URL) {
+        const coverityCommandFormatter = new SynopsysToolsParameter(tempDir);
+        formattedCommand = formattedCommand.concat(
+          coverityCommandFormatter.getFormattedCommandForCoverity()
+        );
+      }
+
       let validationErrors: string[] = [];
       validationErrors = validationErrors.concat(polarisErrors);
```
[the remark this answers is not in the document]
Comment (spurohitsynopsys) [reply synopsys-task/src/synopsys-task/synopsys-bridge.ts]: added

File: synopsys-task/src/synopsys-task/validator.ts (hunk 3 above)
Comment (spurohitsynopsys) [reply synopsys-task/src/synopsys-task/validator.ts]: changed with taskilb exist function

**Review by kishorikumar — APPROVED** [review]

**Pull request merged**